SED-1689 remove only invalid styles, instead of complete css

saem opened the pull request.

teji885 committed:
SED-1689 remove only invalid styles, instead of complete css

saem commented on lib/style_parser/filterer/index.js:
```diff
@@ -20,16 +20,30 @@ var css = require('css');
-    // for now just remove this css until we have a more comprehensive solution to auto-fix
-    return cb(null, '');
+    // for now we are just removing the line that has invalid css, ideally we should auto fix it
+    for( var i=0; i<errors.length; ++i) {
```
This is a really neat idea, I like it.

saem commented on lib/style_parser/filterer/index.js:
```diff
@@ -20,16 +20,30 @@ var css = require('css');
 function filter(cssInput, cb) {
   // Remove HTML comment start/end
   cssInput = removeHtmlComments(cssInput);
+  cssInput = handleParsingErrors(cssInput);
```
`handleParsingErrors` already parses the sheet, is there a way to avoid doing that a second time?

tajinder replied:
handleParsingErrors parses the input css which might have errors.
we will have to parse it second time on css which has errors removed, so that the parsed sheet has no parsing errors.

tajinder replied:
I have changed my code, now it doesn't need t parse the css 2 times

saem commented on lib/style_parser/filterer/index.js:
```diff
@@ -20,16 +20,30 @@ var css = require('css');
+    // for now we are just removing the line that has invalid css, ideally we should auto fix it
+    for( var i=0; i<errors.length; ++i) {
+      if(errors[i].line) {
+        cssInput = cssInput.split("\n").slice(errors[i].line).join("\n")
```
I don't know if it's a good idea to mutate the input parameter.

tajinder replied:
I don't think it should cause any issue

teji885 committed:
SED-1689 instead of looking for lines with errors in input css now removing all rules which have undefined declarations
undefined declarations are created by invalid style in css input, and these rules with undefined declarations cause error with stringify

teji885 committed:
update package version to 0.8.1
